### DISCUSSION TITLE
Can't get API rule to work (array ?= value)

### DISCUSSION THREAD

**Discussion by 98mux** [pocketbase/pocketbase#4689] [Q&A] [answered]
I'm probably just being silly somewhere but I've tried for a whole day to get this api rule to work.

I have a table called "Chat" that has a collection of userIds of all the users who are within the chat.

All of the 5 API rules are set to this:
```userIds ?= @request.auth.id```
Where the requester must be one of the users in userIds

Thought everytime I call this I just get an empty array (getList)
If i remove the rule from List & View rules then it works (I get a non-empty array)
I'm pretty sure I'm properly logged in, as it works on the other tables with ```@request.auth.id = userId```

Grateful for any help 😃

**Comment by ganigeorgiev** [comment, marked as answer]
If `userIds` is a multiple `relation` field then you can try with `userIds.id ?= @request.auth.id` or `userIds:each ?= @request.auth.id` (_the `:each` in the latter translates to the SQLite `json_each` builtin function_).

**Reply by 98mux** [reply]
Again, thank you so much! ```userIds.id ?= @request.auth.id``` worked!!! :heart:
